docs/proposals/25-namespaced-cloud-profiles.md

```diff
@@ -0,0 +1,343 @@
+
+Suppose we have a simplified `CloudProfile` that looks like this:
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: CloudProfile
+metadata:
+  name: aws-central-cloud-profile
+spec:
+  type: aws
+  kubernetes:
+    versions:
+      - version: 1.27.1
+      - version: 1.26.3
+      - version: 1.25.8
+      - version: 1.24.6
+      - version: 1.28.6
+        expirationDate: 2023-02-02T01:02:03Z
+  machineImages:
+    - name: suse-chost
+      versions:
+        - version: 15.4
+        - version: 14.4
+        - version: 13.6
+  machineTypes: 
+    - name: m5.large
+      cpu: "4"
+      gpu: "0"
+      memory: 8Gi
+  volumeTypes:
+    - name: gp3
+      class: standard
+      usable: true
+```
+
+and the `NamespacedCloudProfile` from the [manifest section](#manifest).
+
+After the rendering is done, the `NamespacedCloudProfile` will look like this:
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: NamespacedCloudProfile
+metadata:
+  name:  aws-profile-xyz
+  namespace: project-xyz
+spec:
+  parent:
+    kind: CloudProfile
+    name: aws-central-cloud-profile
+  kubernetes:
+    versions:
+      - version: 1.28.6
+        expirationDate: 2024-06-06T01:02:03Z
+  machineImages:
+    - name: suse-chost
+      versions:
+        - version: 16.4
+          expirationDate: 2023-08-8T23:59:59Z
+  machineTypes: 
+    - name: m5.xlarge
+      cpu: "8"
+      gpu: "0"
+      memory: 16Gi
+  volumeTypes:
+    - name: ab6
+      class: premium
+      usable: true
+status:
+  cloudProfile:
+    apiVersion: core.gardener.cloud/v1beta1
+    kind: CloudProfile
+    spec:
+      type: aws
+      kubernetes:
+        versions:
+          - version: 1.27.1
+          - version: 1.26.3
+          - version: 1.25.8
+          - version: 1.24.6
+          - version: 1.28.6
+            expirationDate: 2024-06-06T01:02:03Z
+      machineImages:
+        - name: suse-chost
+          versions:
+            - version: 16.4
+              expirationDate: 2023-08-8T23:59:59Z
+            - version: 15.4
+            - version: 14.4
+            - version: 13.6
+      machineTypes: 
+        - name: m5.large
+          cpu: "4"
+          gpu: "0"
+          memory: 8Gi
+        - name: m5.xlarge
+          cpu: "8"
+          gpu: "0"
+          memory: 16Gi
+      volumeTypes:
+        - name: ab6
+          class: premium
+          usable: true
+        - name: gp3
+          class: standard
+          usable: true
+```
+
+The rendering is done by a new custom controller registered to the `gardener-controller-manager`. Merge conflicts can not arise during the merge process as they are caught by static validation and an admission plugin for validating the `NamespacedCloudProfile` object.
```

Comment:
🤔 does this mean a `NamespacedCloudProfile` can disallow an update of the parent `CloudProfile` that would introduce a merge conflict?